lib/class-wp-widget-block.php

```diff
@@ -53,11 +53,92 @@ public function __construct() {
 	 * @global WP_Post $post Global post object.
 	 */
 	public function widget( $args, $instance ) {
-		echo $args['before_widget'];
+		echo sprintf( $args['before_widget'], $this->get_dynamic_classname( $instance ) );
```

Comment:
It would have been nice to determine the `classname` in the constructor and not have to use `sprintf` here. It works, to get the instance in the constructor but I didn't test with all kinds of widgets.

Reply:
We can't determine the `classname` in the constructor because the block HTML is in `$instance`.

Reply:
> It works, to get the instance in the constructor but I didn't test with all kinds of widgets.

How did you get the `$instance`? Could you share a code snippet?

Reply:
The same way it is obtained in `WP_Widget::display_callback`